### ISSUE TITLE
Relation to qutebrowser

### ISSUE THREAD

**Issue by khinsen** [parkouss/webmacs#3]
I wonder if you are aware of [qutebrowser](https://www.qutebrowser.org/), a very similar project (in particular using Python + Qt). At a first glance, your goal seems to be to do the same but with Emacs-inspired bindings rather than qutebrowser's vi-inspired user interface. Could webmacs become an alternative user interface for qutebrowser, to avoid duplicating lots of lower-level work that another project has already done?

**Comment by parkouss** [comment]
First yes, I am aware of qutebrowser - note I was not when I started the project. It is indeed quite similar (meaning keyboard-based and Python + Qt/webengine).

If I understand what you are saying, you are wondering if webmacs could become a plugin for qutebrowser. Honnestly I don't know about that, I don't know if there is a plugin system for qutebrowser and if there is if it would be sufficient for that.

As far as I can see, qutebrowser attracts lots of people, so maybe emacs bindings will become available and ready to use without plugin at some point. I believe there are some other differences right now, maybe fuzzy matching in many places, webjump concept, ability to split the window in multiple views, a fast ad-blocker... (I might miss some or be wrong on others!)

I tried a few weeks ago to build a qutebrowser configuration that would fit my workflow, but without luck (if you have some hints or a config emacs-like, please share it).

I think I understand your point of view. For now, webmacs just better address what I want (being used to the conkeror way of browsing) and it basically fill my needs, and I thought others could find it useful so I just shared it on github. It would be good if some core code could be shared between the two projects, or maybe if qutebrowser allowed the same workflow with configuration, but I must say it sounds like more work than what I can give on this.

**Comment by The-Compiler** [comment]
qutebrowser's author here :wink:

I know some ex-conkeror users which switched to qutebrowser and are happy with it so far. I think the missing features to make them feel at home is a way to chain keybindings with modifiers (https://github.com/qutebrowser/qutebrowser/issues/319) and a way to disable insert mode (https://github.com/qutebrowser/qutebrowser/issues/3143).

There's no plugin API yet (though that's planned, https://github.com/qutebrowser/qutebrowser/issues/30), but as for allowing emacs-like keybindings and stuff, I'd rather have a generic way of doing so in the core (by looking at the issues above). Always happy about more feedback from conkeror users though, even though I can't promise anything as it takes a lot of time! 😄 

> maybe fuzzy matching in many places

Depends on what fuzzy-matching you mean exactly, but the completion (e.g. for `:open`/`o`) does do fuzzy-matching, and it's been recently improved in the `master` branch. However, `foobar` won't match `foo baz bar` (but `foo bar` will).

> webjump concept

If I'm understanding webjumps right, that's available, as the `url.searchengines` setitng.

> ability to split the window in multiple views

That's something qutebrowser can't do yet, and I'm not sure whether it's worth the complexity as most users use a tiling window manager anyways. I plan to take a look at how webmacs implemented that, though! :laughing:

> a fast ad-blocker

That is available, but right now it only checks against blacklisted hosts, nothing more sophisticated.

> I tried a few weeks ago to build a qutebrowser configuration that would fit my workflow, but without luck (if you have some hints or a config emacs-like, please share it).

I'm wondering, anything that was going wrong other than the two issues I listed above?

**Comment by parkouss** [comment]
> I'm wondering, anything that was going wrong other than the two issues I listed above?

Nothing wrong, I tried it and did not feel at home (I installed it twice and tried to configure it for one hour or something like that). I already had started to write webmacs at that time and it was quickier to just add what was missing in webmacs (again, for me to feel at home).

Also I was excited to write my own browser. :)

I am closing this issue - I believe even if qutebrowser and webmacs are made with the same technologies, they are quite different beast and QtWebEngine is globally what make the dirty work here, and it is already shared between the projects.
